app/Contact.php

```diff
@@ -1078,12 +1079,12 @@ public function getIntroducer()
      */
     public function setSpecialDate($occasion, int $year, int $month, int $day)
     {
-        if (is_null($occasion)) {
+        if (null === $occasion) {
             return;
         }
 
         $specialDate = new SpecialDate;
-        $specialDate->createFromDate($year, $month, $day)->setToContact($this);
+        $specialDate->setToContact($this)->createFromDate($year, $month, $day);
```

Comment:
Why do you change the order?
With your solution the specialDate is saved before it's created. What was wrong with how it was done before?

Reply:
Because you cannot create a special date without specifying the associated contact/account :stuck_out_tongue_winking_eye:.

However I'd prefer to get the calls to `save` out of the model so that the order would not matter. And also have `createFromDate` not do `save`.

This kind of logic should go into a service (a class dedicate to a specific purpose).

Reply:
Interesting. Thanks a lot for your patience in explaining all this to me. As you can see, I don't have your knowledge nor skills 😀